server/controllers/ssotVariableController.js

```diff
@@ -0,0 +1,335 @@
+                activityId: activityParameterId,
+            },
+            { 
+                rpaParameters: updatedParameters,
+                outputVariable: `${activityParameterId}_output`
+            },
+            {
+                new: true,
+                useFindAndModify: false,
+                upsert: true
+            }
+        ).exec()
+    } 
+    
+    return mongoose.model('parameter').findOneAndUpdate(
+        {
+            ssotId: botId,
+            activityId: activityParameterId,
+        },
+        { 
+            rpaParameters: updatedParameters,
+            $unset: { outputVariable: 1 }
+        },
+        {
+            new: true,
+            useFindAndModify: false,
+            upsert: true
+        }
+    ).exec()    
+}
+
+const checkForOutputValue = async (botId, activityParameterId) => {
+    const parameterObject = await  mongoose.model('parameter').findOne(
+        {
+            ssotId: botId,
+            activityId: activityParameterId,
+        }
+    ).exec();
+    return !!parameterObject.outputVariable
+}
+
+const updateParametersForActivity = async (botId, activityParameterId, updatedParameters, updatedOutput, hasOutput) => {
+    if (hasOutput) {
+        return mongoose.model('parameter').findOneAndUpdate(
+            {
+                ssotId: botId,
+                activityId: activityParameterId,
+            },
+            { 
+                rpaParameters: updatedParameters,
+                outputVariable: updatedOutput
+            },
+            {
+                new: true,
+                useFindAndModify: false,
+                upsert: true
+            }
+        ).exec()
+    } 
+    
+    return mongoose.model('parameter').findOneAndUpdate(
+        {
+            ssotId: botId,
+            activityId: activityParameterId,
+        },
+        { 
+            rpaParameters: updatedParameters,
+            $unset: { outputVariable: 1 }
+        },
+        {
+            new: true,
+            useFindAndModify: false,
+            upsert: true
+        }
+    ).exec()    
+}
+
+// GET /ssot/getVariablesForNewTask/?botId=6045eccf&activityId=ActivityId123&application=MS+Excel&task=Open+Application
+exports.getVariablesForNewTask = async (req, res) => {
+    try {
+        res.set('Content-Type', 'application/json');
+        const { botId } = req.query;
+        const { activityId } = req.query;
+        const { application } = req.query;
+        const applicationWithEmptyspace = application.replace(/\+/g, ' ');
+        const { task } = req.query;
+        const taskWithEmptyspace = task.replace(/\+/g, ' ');
+
+        const rpaTaskVariables = await findParametersForTask(applicationWithEmptyspace, taskWithEmptyspace);
+
+        const inputVars = [];
+        rpaTaskVariables.get('inputVars').forEach((singleVar) => {
+            const varWithValue = singleVar;
+            varWithValue.value = '';
+            inputVars.push(varWithValue);
+        });

```

Comment:
I would rather use 'parameter' concistently and not switch with the naming between variables and parameters.

Reply:
Same goes for all the following functions having 'variables' in their name and also in the functions body

Reply:
Please explain this to me, as I currently do not get what you mean. We are using the term variables when talking about parameters (input), all well as the possible output value.
An exception from the rule currently ist the inputVars because they are being called that way in the mongoose models and to keep consistency, I kept that naming here too.
Changing the name of the variable is a good point, but we did not change that in this PR and it would drag along many small additional changes. You can (or I) open up a new issue for that, but it should not be part of this PR for now.

Reply:
Okay. Then let's open a new Issue for this one.